conbench/app/_plots.py

```diff
@@ -399,17 +415,26 @@ def time_series_plot(history, benchmark, run, height=380, width=1100):
     unit = history[0]["unit"]
     with_dist = [h for h in history if h["distribution_mean"]]
     formatted, axis_unit = _should_format(history, unit)
+    dist_change_ixs = [
+        ix
+        for ix, result in enumerate(history)
+        if result["change_annotations"].get("begins_distribution_change", False)
+    ]
 
     # Note(JP): `history` is an ordered list of dicts, each dict has a `mean`
     # key which is extracted here by default.
     source_mean_over_time = _source(history, unit, formatted=formatted)
 
     source_min_over_time = bokeh.models.ColumnDataSource(
         data=dict(
-            x=util.tznaive_iso8601_to_tzaware_dt([x["timestamp"] for x in history]),
+            # Insert NaNs to break the line at distribution changes
+            x=_insert_nans(
+                util.tznaive_iso8601_to_tzaware_dt([x["timestamp"] for x in history]),
+                dist_change_ixs,
+            ),
             # TODO: best-case is not always min, e.g. when data has a unit like
             # bandwidth.
-            y=[min(x["data"]) for x in history],
+            y=_insert_nans([min(x["data"]) for x in history], dist_change_ixs),
```

Comment:
I like the user-facing change that this PR brings. This implementation complexity of this module and also the readability might slow some people down in the future, but we can tweak this later.

Reply:
+100